api/src/test/java/com/exelaration/abstractmemery/controllers/UploadControllerTest.java

```diff
@@ -0,0 +1,65 @@
+package com.exelaration.abstractmemery.controllers;
+
+import static org.mockito.Mockito.when;
+
+import com.exelaration.abstractmemery.domains.Image;
+import com.exelaration.abstractmemery.services.implementations.ImageServiceImpl;
+import java.io.File;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = UploadController.class)
+public class UploadControllerTest {
+
+  private MockMvc mockMvc;
+  @Autowired private UploadController uploadController;
+  @MockBean private ImageServiceImpl imageService;
+
+  @BeforeEach
+  public void setUp() {
+    this.mockMvc = MockMvcBuilders.standaloneSetup(uploadController).build();
+  }
+
+  @Test
+  public void testSaveImageReturns200() throws Exception {
```

Comment:
Test names should be specific enough to convey what's being tested and what you expect.  A good rule of thumb is to use the format `FunctionUnderTest_When<SetupConditions>_Expect<Result>`